tests/Fixtures/TestBundle/GraphQl/Type/Definition/DateTimeType.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Tests\Fixtures\TestBundle\GraphQl\Type\Definition;
+
+use ApiPlatform\Core\GraphQl\Type\Definition\TypeInterface;
+use GraphQL\Error\Error;
+use GraphQL\Language\AST\StringValueNode;
+use GraphQL\Type\Definition\ScalarType;
+use GraphQL\Utils\Utils;
+
+/**
+ * Represents a DateTime type.
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class DateTimeType extends ScalarType implements TypeInterface
+{
+    public function __construct()
+    {
+        $this->name = 'DateTime';
+        $this->description = 'The `DateTime` scalar type represents time data.';
+
+        parent::__construct();
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function serialize($value)
+    {
+        // Already serialized.
+        if (\is_string($value)) {
+            // Should be better in a custom normalizer.
+            return (new \DateTime($value))->format('Y-m-d');
```

Comment:
Definitely, shouldn't you inject the DateTimeNormalizer?

Reply:
Why? The comment is here to warn the user that using a custom normalizer is a better idea that using a custom type.
But I need to test the custom type to. I you have a better example for using a custom type, you can tell me because this one is not really great.

Reply:
Oh I understand now thanks. I'd have used UUIDs as custom types for example but it's not a big deal.

Reply:
Is there something to manage with UUID? Because IMO it's just a string doesn't it? Maybe just the validation?

Reply:
no you could transform it into a Ramsey\Uuid object?

Reply:
Hmm I see. I could try.

Reply:
It seems there are other issues with Ramsey UUID (`IdentifierConverter` is never called) so I'm keeping this test for now.